FreeRTOS_Sockets.c

```diff
@@ -3394,7 +3394,7 @@ void vSocketWakeUpUser( FreeRTOS_Socket_t * pxSocket )
                               size_t uxDataLength,
                               BaseType_t xFlags )
     {
-        BaseType_t xByteCount = -pdFREERTOS_ERRNO_EINVAL;
+        BaseType_t xByteCount;
```

Comment:
The comment in line 3385 should also be updated to reflect that the buffer can be NULL or non-NULL and the conditions can be added in a `@note` section.

Reply:
Yes, that we can do, I am proposing:
~~~c
/*
 * @param[in] pvBuffer: The buffer containing the data. The value of this pointer
 *                      may be NULL in case zero-copy transmissions are used.
 *                      It is used in combination with 'FreeRTOS_get_tx_head()'.
 */
~~~

Reply:
Yes, that would be very helpful, thank you.

Reply:
> Yes, that would be very helpful, thank you.

I added the comment, and also in the IPv6/multi version of this PR: #317

Reply:
Hello! do you mind pushing the changes so that we can get this merged into the repository?

Thanks :)

Reply:
It got stuck in an unpushed commit. Now it it there.